test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/ZookeeperServiceOnK8.java

```diff
@@ -0,0 +1,244 @@
+                                               .thenCompose(v -> k8Client.createClusterRole(getClusterRole()))
+                                               .thenCompose(v -> k8Client.createClusterRoleBinding(getClusterRoleBinding()))
+                                               //deploy zk operator.
+                                               .thenCompose(v -> k8Client.createDeployment(NAMESPACE, getDeployment()))
+                                               // wait until zk operator is running, only one instance of operator is running.
+                                               .thenCompose(v -> k8Client.waitUntilPodIsRunning(NAMESPACE, "name", OPERATOR_ID, 1))
+                                               // request operator to deploy zookeeper nodes.
+                                               .thenCompose(v -> k8Client.createAndUpdateCustomObject(CUSTOM_RESOURCE_GROUP, CUSTOM_RESOURCE_VERSION, NAMESPACE, CUSTOM_RESOURCE_PLURAL, getZookeeperDeployment(getID(), 1))), t -> new TestFrameworkException(RequestFailed, "Failed to deploy zookeeper operator/service", t));
+        if (wait) {
+            Futures.getAndHandleExceptions(k8Client.waitUntilPodIsRunning(NAMESPACE, "app", ID, 1),
+                                           t -> new TestFrameworkException(RequestFailed, "Failed to deploy zookeeper service", t));
+        }
+    }
+
+
+    @Override
+    public void stop() {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public void clean() {
+    }
+
+    @Override
+    public String getID() {
+        return ID;
+    }
+
+    @Override
+    public boolean isRunning() {
+        // return false since mutliple invocations of start does not have any side effects.
+        return false;
+    }
+
+    @Override
+    public List<URI> getServiceDetails() {
+        //fetch the URI.
+        return Futures.getAndHandleExceptions(k8Client.getStatusOfPodWithLabel(NAMESPACE, "app", ID)
+                                                      .thenApply(statuses -> statuses.stream().map(s -> URI.create(TCP + s.getPodIP() + ":" + ZKPORT))
+                                                                                     .collect(Collectors.toList())),
+                                              t -> new TestFrameworkException(RequestFailed, "Failed to fetch ServiceDetails for Zookeeper", t));
+    }
+
+    @Override
+    public CompletableFuture<Void> scaleService(int instanceCount) {
+        //Update the instance count.
+        // request operator to deploy zookeeper nodes.
+        return k8Client.createAndUpdateCustomObject(CUSTOM_RESOURCE_GROUP, CUSTOM_RESOURCE_VERSION, NAMESPACE, CUSTOM_RESOURCE_PLURAL,
+                                             getZookeeperDeployment(getID(), instanceCount))
+                .thenCompose(v -> k8Client.waitUntilPodIsRunning(NAMESPACE, "app", ID, instanceCount));
+    }
+
+    private V1beta1ClusterRoleBinding getClusterRoleBinding() {
```

Comment:
If you change cluster-wide roles to regular roles, make sure to update the binding as well: `ClusterRoleBinding` -> `RoleBinding`

Reply:
Sure. Currently we are using default namespace for all the operators and tests, hence used ClusterRoleBinding https://github.com/pravega/zookeeper-operator/blob/master/deploy/default_ns/rbac.yaml